### PR TITLE
Fix application window not listed in taskbar (X11)

### PR DESCRIPTION
**Pull request by MJacred**
* for exported executable
* for editor and exported executable (on older Cinnamon versions)

### CODE REVIEW

File: platform/x11/context_gl_x11.cpp [hunk 1]
```diff
@@ -191,7 +191,8 @@ Error ContextGL_X11::initialize() {
 
 	swa.colormap = XCreateColormap(x11_display, RootWindow(x11_display, vi->screen), vi->visual, AllocNone);
 	x11_window = XCreateWindow(x11_display, RootWindow(x11_display, vi->screen), 0, 0, OS::get_singleton()->get_video_mode().width, OS::get_singleton()->get_video_mode().height, 0, vi->depth, InputOutput, vi->visual, valuemask, &swa);
-
+	XStoreName(x11_display, x11_window, "Godot Engine");
```
Comment (Sslaxx) [inline platform/x11/context_gl_x11.cpp]: Forgive my lack of knowledge, but doesn't this mean that exported applications would also show up as "Godot Engine" on the taskbar?

Comment (MJacred) [reply platform/x11/context_gl_x11.cpp]: XStoreName() is also called each time the title of window is changed and therefore will be overwritten immediately (I checked). But not calling it at this point causes the issue...

Comment (Sslaxx) [reply platform/x11/context_gl_x11.cpp]: Ah, gotcha. Thanks for explaining!

File: platform/x11/context_gl_x11.cpp [hunk 2]
```diff
@@ -191,7 +191,8 @@ Error ContextGL_X11::initialize() {
...
+
```
Comment (akien-mga) [inline platform/x11/context_gl_x11.cpp]: Not sure why GitHub doesn't make it selectable now, but there's a trailing tab here that should be removed.

File: platform/x11/context_gl_x11.cpp (hunk 1 above)
Comment (MJacred) [reply platform/x11/context_gl_x11.cpp]: But you made me curious for the Window-Preview. So I checked. Here it doesn't update without a workspace switch... But that's on my old Cinnamon. Someone with newer version or with Gnome needs to test this...

godot_executable

**Comment by akien-mga** [comment]
See http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase for help on rebase your branch :)

**Comment by MJacred** [comment]
> See http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase for help on rebasing your branch :)

I'm trying... but it seems to not want to do the fixup properly... Maybe I'll sleep over this and try tomorrow again (and git has some serious merge conflict issues with removing tabs-only in a line...)

**Comment by MJacred** [comment]
OK, now I got it (it was a series of merge conflicts and I had to meld them one after the other and overwrite unwanted results. 1UP! on a better understanding of merge conflicts). Now I'm kinda glad and a little proud. I was afraid I had to scrap the fork and redo it from scratch ٩( ๑╹ ꇴ╹)۶

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**